.github/workflows/pr_ci.yml

```diff
@@ -15,17 +15,15 @@ jobs:
     name: Build & Test
     steps:
     - uses: actions/checkout@v2
-    - uses: actions/setup-node@v2
+    - uses: actions/cache@v3
       with:
-        node-version: "15"
-    - name: Setup .NET 6.0
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: 6.0.x
-        include-prerelease: true
```

Comment:
v3 of `actions/setup-dotnet` already includes a feature not to install a given version if it already exists on the VM. So, using it should be enough for us to use the version we want if it exists on the image but still get the speedup: https://github.com/Lombiq/GitHub-Actions/pull/67/files#diff-7978921ed7153ba992ebf38332bdd64fe6f3acb10a452b2978859c5a835ea74bR30 (note the comment above, I initially understood the behavior differently).

Reply:
Ideally build system would handle this. For example NUKE checks solution's global.json for required SDK version and downloads if needed. Works also outside of GitHub. But probably upgrading actions is the way to go when we can't use the best tools..

Reply:
Well, we can have something fancy, but in the short term, just pointing this action to v3 will optimize this :).

Reply:
Just had a run using v3 and it took 1m 29s, so it still comes with a price...

![image](https://user-images.githubusercontent.com/171892/196026119-4bff20fe-cc69-4be0-ac91-983c0907204c.png)

Reply:
That's supposed to be a lot faster... In practice I've seen <20s for this (vs ~3 minutes for a full install). So, I don't know.

Reply:
The build you're showing was perhaps a one-time fluke. I again checked our builds, and these restores are ~15s when using a preinstalled SDK. See e.g.:

- https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/3257120788/jobs/5348182484#step:3:1
- https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/3257120788/jobs/5348182424#step:3:1
- https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/3256910188/jobs/5347817613#step:3:1
- https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/3256910188/jobs/5347817671#step:3:1
- https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/3257569220/jobs/5348910654#step:3:1
- https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/3257569220/jobs/5348910715#step:3:1

Reply:
15 seconds is quite long time for no-op?

Reply:
That's what I initially thought as well, but [as someone kindly explained in the mentioned thread](https://github.com/actions/setup-dotnet/issues/284#issuecomment-1276250960) it's not a no-op :). It's also checking the latest released `6.0.x` version online, and if it's newer than the pre-installed one, installs that instead.